src/integ_test_resources/ios/sdk/integration/cdk/custom_resources/iot_custom_authorizerUserPass_provider/iot_custom_authorizerUserPass_provider.py

```diff
@@ -0,0 +1,107 @@
+    public_key = event["ResourceProperties"]["public_key"]
+    token_key_name = event["ResourceProperties"]["token_key_name"]
+    create_authorizer(
+        custom_authorizer_name, custom_authorizer_function_arn, public_key, token_key_name
+    )
+
+    #Creating a domain config is required for enhanced user auth
+    beta_endpoint_addr = create_or_update_domainConfig()
+    response = {
+        "PhysicalResourceId": physical_id,
+        "Data": {"BetaEndpointAddress": beta_endpoint_addr}
+    }
+    print(f"### on_create_or_update response: {response}")
+    return response
+
+def create_or_update_domainConfig():
+    client = boto3.client("iot")
+
+    try:
+        create_domain_configuration_response = client.create_domain_configuration(
+            domainConfigurationName=domain_configuration_name)
+        print(f"### create_domain_configuration response: {create_domain_configuration_response}")
+    except client.exceptions.ResourceAlreadyExistsException as e:
+        print("Already defined, continuing")
+
+    update_domain_configuration_response = client.update_domain_configuration(
+        domainConfigurationName=domain_configuration_name,
+        domainConfigurationStatus='ENABLED',
+        authorizerConfig={'defaultAuthorizerName': default_authorizer_name})
+    print(f"### update_domain_configuration response: {update_domain_configuration_response}")
+
+    describe_domain_configuration_response = client.describe_domain_configuration(
+        domainConfigurationName=domain_configuration_name)
+    print(f"### describe_domain_configuration response: {describe_domain_configuration_response}")
+
+    beta_endpoint_addr = describe_domain_configuration_response["domainName"]
+    print(f"### beta_endpoint_addr: {beta_endpoint_addr}")
+
+    return beta_endpoint_addr
+
+def on_delete(event):
+    print(f"### on_delete({event})")
+    custom_authorizer_name = event["ResourceProperties"]["authorizer_name"]
+    delete_authorizer(custom_authorizer_name)
+    response = {"PhysicalResourceId": physical_id}
+    print(f"### on_delete response: {response}")
+    return response
+
+
+def create_authorizer(name, function_arn, public_key, token_key_name):
+    print(f"### create_authorizer({name, function_arn, public_key, token_key_name})")
+    client = boto3.client("iot")
+    formatted_key = format_public_key(public_key)
+    create_authorizer_response = client.create_authorizer(
+        authorizerName=name,
+        authorizerFunctionArn=function_arn,
+        tokenKeyName=token_key_name,
+        status="ACTIVE",
+        signingDisabled=False,
+        tokenSigningPublicKeys={"public_key": formatted_key},
+    )
+    print(f"### create_authorizer_response: {create_authorizer_response}")
+
+
+def format_public_key(public_key):
+    return f"-----BEGIN PUBLIC KEY-----\n{public_key}\n-----END PUBLIC KEY-----"
+
+
+def delete_authorizer(name):
+    print(f"### delete_authorizer({name})")
+    client = boto3.client("iot")
```

Comment:
I'd call this `iot` instead of `client`, maybe.

Reply:
All of the other code in the IoT Stack refers to to the IoT boto client as just "client" if we were to change this, we'd probably want to be consistent in the entire stack/package... for now, i'm going to leave this as-is.